### PR TITLE
feat: Introduce new version v1 for using in A2A integration

### PR DESCRIPTION
**Pull request by drobnikj**

### CODE REVIEW

File: .actor/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Specify the base Docker image
+FROM oven/bun:1.0
```
Comment (valekjo) [inline .actor/Dockerfile]: 👍 I think for the "fast things without migrations" it works fine.

File: .actor/input_schema.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            "editor": "textfield"
+        },
+        "uniqueKeys": {
+            "title": "Unique attributes to identify a document",
+            "type": "array",
+            "description": "Unique keys for object, if you specified unique keys, act try to find object with this attributes in DB and update it.",
+            "editor": "json"
+        }
+    },
+    "required": ["mongoUrl", "datasetId"]
```
Comment (valekjo) [inline .actor/input_schema.json]: Suggestion: Maybe "collection" should be required too?

File: package.json [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
+    "mongodb": "latest"
+  },
+  "devDependencies": {
+    "@apify/eslint-config-ts": "^0.4.0",
+    "@apify/tsconfig": "^0.1.0",
+    "@typescript-eslint/eslint-plugin": "^6.9.1",
+    "@typescript-eslint/parser": "^6.9.1",
+    "bun-types": "latest",
+    "eslint": "^8.52.0",
+    "typescript": "^5.2.2"
```
Comment (valekjo) [inline package.json]: Question: Are those needed with bun?

File: src/main.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import { Actor, log } from 'apify';
+import { MongoClient } from 'mongodb';
+
+import { ImportStats, Types } from './types';
+import { importFromDataset } from './utils';
+
+const DEFAULT_COLLECTION_NAME = 'results';
```
Comment (valekjo) [inline src/main.ts]: Suggestion: Maybe I'd move this to input schema then (as default).

File: src/utils.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+export function pickFromObject<T extends object, K extends keyof T>(object: T, keys: K[]): Partial<T> {
+    return keys.reduce((obj: Partial<T>, key: K) => {
+        if (object && Object.prototype.hasOwnProperty.call(object, key)) {
+            obj[key] = object[key];
+        }
+        return obj;
+    }, {});
+}
+
+export async function importObjectToCollection(object: Record<string, unknown>, { collection, importStats, uniqueKeys }: ImportOptions) {
```
Comment (valekjo) [inline src/utils.ts]: Suggestion: Maybe it would be worth to use `bulkWrite` in this case?

File: .actor/input_schema.json (hunk 2 above)
Comment (drobnikj) [reply .actor/input_schema.json]: There is default collection used in this case.

**Comment by drobnikj** [comment]
Little big forget about this one, can you do one more round 😄 @valekjo

File: package.json (hunk 3 above)
Comment (valekjo) [reply package.json]: I think we only need `bun-types` as a dependency. Or not?

**Review by valekjo — APPROVED** [review]
Let's release it like this 👍

**Comment by OPesicka** [comment]
Ping. We would like to use this for A2A promotion

**Pull request merged**